src/Entity/EntityDocument.php

```diff
@@ -31,4 +33,15 @@ public function getId();
 	 */
 	public function getType();
 
+	/**
+	 * Sets the id of the entity. If the id is not of the correct type,
+	 * an exception will be thrown. A specific derivative of EntityId is
+	 * always supported.
+	 *
+	 * @param EntityId|mixed $id
```

Comment:
That depends on how you interpret `mixed`. If you interpret it as "must support all types", then yes.

I'm interpreting this one as "and possibly something else". I started with just having `EntityId` as type hint, though then you'd get IDE warnings... ... no wait, you only will when not using the subtype, in which case you're doing it wrong... so yeah, should just be dropped then.